### PR TITLE
SOAR-1867 Add new workflow named Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams

### PR DESCRIPTION
**Pull request by ekim-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1
workflow2

#### Job

job1

#### Artifact

teams1
teams2
Screen Shot 2020-09-01 at 16 08 14
Screen Shot 2020-09-01 at 16 08 55

### CODE REVIEW

**Comment by ekim-r7** [comment]
% icon-validate .
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 23.188ms

**Comment by ncramer-r7** [comment]
In terms of the workflow, opt for using the automatic indicator extraction (aka Quarantine Request.message.first_word) instead of the pattern match step when possible

File: workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon [hunk 1]
```diff
@@ -0,0 +1,2014 @@
...
+  "kom": {
+    "komandVersion": "1.60.3-0-g2bc27",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-09-01T20:13:33.501123327Z",
+    "workflowVersions": [
+      {
+        "name": "Quarantine Endpoint with Microsoft Defender ATP from Microsoft Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "",
```
Comment (jmcadams-r7) [inline workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon:11]: Need a description

File: workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon [hunk 2]
```diff
@@ -0,0 +1,2014 @@
...
+              "slugName": "microsoft_teams",
+              "slugVersion": "2.2.0",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.2.0/icon.png"
+            },
+            "identifier": "new_message_received",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "integration-testing",
```
Comment (jmcadams-r7) [inline workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon:589]: need to change the channel name and team name to "change_me"

File: workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/workflow.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+extension: workflow
+products: ["insightconnect"]
+name: Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams
+title: "Quarantine Endpoint with Microsoft Defender ATP from Microsoft Teams"
+description: "Quarantine (lockdown) an endpoint in Microsoft Defender ATP from a Microsoft Teams command."
```
Comment (jmcadams-r7) [inline workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/workflow.spec.yaml:5]: Remove (lockdown). That's specific to another product.

**Review by jmcadams-r7 — DISMISSED** [review]

File: workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon [hunk 4]
```diff
@@ -0,0 +1,2014 @@
...
+            "type": "action",
+            "plugin": {
+              "name": "Microsoft Windows Defender ATP",
+              "slugVendor": "rapid7",
+              "slugName": "microsoft_atp",
+              "slugVersion": "4.4.1",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_atp/4.4.1/icon.png"
+            },
+            "identifier": "isolate_machine",
+            "continueOnFailure": false,
```
Comment (ncramer-r7) [inline workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon:211]:
```suggestion
            "continueOnFailure": true,
```

From the looks of the wf, I think you probably want to toggle this to true. That way, you'll be able to hit the "Not Successful" branch

File: workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon [hunk 5]
```diff
@@ -0,0 +1,2014 @@
...
+            "type": "action",
+            "plugin": {
+              "name": "Microsoft Windows Defender ATP",
+              "slugVendor": "rapid7",
+              "slugName": "microsoft_atp",
+              "slugVersion": "4.4.1",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_atp/4.4.1/icon.png"
+            },
+            "identifier": "unisolate_machine",
+            "continueOnFailure": false,
```
Comment (ncramer-r7) [inline workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon:1591]:
```suggestion
            "continueOnFailure": true,
```

Same reasoning as above

File: workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon [hunk 6]
```diff
@@ -0,0 +1,2014 @@
...
+              "title": "Variables",
+              "type": "object"
+            },
+            "outputJSONSchema": null,
+            "defaultImageData": "",
+            "connectionType": "CONNECTION"
+          }
+        },
+        "tags": null,
+        "humanCostSeconds": 0,
```
Comment (ncramer-r7) [inline workflows/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams/Quarantine_Endpoint_with_Microsoft_Defender_ATP_from_Microsoft_Teams.icon:1699]:
```suggestion
        "humanCostSeconds": 300,
```

Just to add some time-saved value

**Review by ncramer-r7 — APPROVED** [review]

**Review by jmcadams-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**